### ISSUE TITLE
Gigabyte BRIX mon server hard drive won't boot after ./securedrop-admin install runs 

### ISSUE THREAD

**Issue by huertanix** [freedomofpress/securedrop#3373]
# Bug

## Description

While installing SecureDrop, mon server is no longer reachable via SSH after running `./securedrop-admin install`. The mon server was previously accessible via SSH after key-based authentication was established as per the docs here: https://docs.securedrop.org/en/stable/servers.html#set-up-ssh-keys. `./securedrop-admin setup` also ran without error, but only after several `./securedrop-admin install` attempts did the mon server's SSD drive stop booting and started displaying this from a TTY at boot:

    Reboot and Select proper Boot device
    or Insert Boot Media in selected Boot device and press a key_

## Steps to Reproduce

Using latest Tails (3.6.2) on an SD Admin USB drive. Went through all steps in SecureDrop documentation here: https://docs.securedrop.org/en/stable/install.html including the prerequisite Ubuntu installs and firewall configurations. `./securedrop-admin install` is ran and an error says the mon server is unreachable. mon server is plugged into a monitor and keyboard and the aforementioned boot error appears.

## Expected Behavior

SecureDrop install runs on app and mon servers successfully. mon server can boot.

## Actual Behavior

This happens when running `./securedrop-admin install`:

    INFO: Now installing SecureDrop on remote servers.
    INFO: You will be prompted for the sudo password on the servers.
    INFO: The sudo password is only necessary during initial installation.
    
    PLAY [Ensure validation is run before prod install] ****************************************************************************
    
    TASK [Gathering Facts] *********************************************************************************************************
    ok: [localhost]
    
    TASK [validate : Confirm host OS is Tails.] ************************************************************************************
    ok: [localhost] => {
        "changed": false, 
        "msg": "All assertions passed"
    }
    
    TASK [validate : Check for persistence volume.] ********************************************************************************
    ok: [localhost] => (item=/live/persistence/TailsData_unlocked/persistence.conf)
    ok: [localhost] => (item=/live/persistence/TailsData_unlocked/openssh-client)
    ok: [localhost] => (item=/home/amnesia/Persistent/securedrop)
    
    TASK [validate : Confirm persistence volume is configured.] ********************************************************************
    ok: [localhost] => (item={'_ansible_parsed': True, u'stat': {u'isuid': False, u'uid': 115, u'exists': True, u'attr_flags': u'', u'woth': False, u'isreg': True, u'device_type': 0, u'mtime': 1525151620.0765016, u'block_size': 4096, u'inode': 12, u'isgid': False, u'size': 527, u'executable': False, u'roth': False, u'charset': u'unknown', u'readable': False, u'version': None, u'pw_name': u'tails-persistence-setup', u'gid': 122, u'ischr': False, u'wusr': True, u'writeable': False, u'isdir': False, u'blocks': 8, u'xoth': False, u'rusr': True, u'nlink': 1, u'issock': False, u'rgrp': False, u'gr_name': u'tails-persistence-setup', u'path': u'/live/persistence/TailsData_unlocked/persistence.conf', u'xusr': False, u'atime': 1525151594.3645008, u'mimetype': u'unknown', u'ctime': 1525151620.0765016, u'isblk': False, u'xgrp': False, u'dev': 65024, u'wgrp': False, u'isfifo': False, u'mode': u'0600', u'islnk': False, u'attributes': []}, u'changed': False, '_ansible_no_log': False, 'item': u'/live/persistence/TailsData_unlocked/persistence.conf', '_ansible_item_result': True, 'failed': False, u'invocation': {u'module_args': {u'checksum_algorithm': u'sha1', u'get_checksum': True, u'follow': False, u'path': u'/live/persistence/TailsData_unlocked/persistence.conf', u'get_md5': True, u'get_mime': True, u'get_attributes': True}}, '_ansible_ignore_errors': None}) => {
        "changed": false, 
        "item": {
            "changed": false, 
            "failed": false, 
            "invocation": {
                "module_args": {
                    "checksum_algorithm": "sha1", 
                    "follow": false, 
                    "get_attributes": true, 
                    "get_checksum": true, 
                    "get_md5": true, 
                    "get_mime": true, 
                    "path": "/live/persistence/TailsData_unlocked/persistence.conf"
                }
            }, 
            "item": "/live/persistence/TailsData_unlocked/persistence.conf", 
            "stat": {
                "atime": 1525151594.3645008, 
                "attr_flags": "", 
                "attributes": [], 
                "block_size": 4096, 
                "blocks": 8, 
                "charset": "unknown", 
                "ctime": 1525151620.0765016, 
                "dev": 65024, 
                "device_type": 0, 
                "executable": false, 
                "exists": true, 
                "gid": 122, 
                "gr_name": "tails-persistence-setup", 
                "inode": 12, 
                "isblk": false, 
                "ischr": false, 
                "isdir": false, 
                "isfifo": false, 
                "isgid": false, 
                "islnk": false, 
                "isreg": true, 
                "issock": false, 
                "isuid": false, 
                "mimetype": "unknown", 
                "mode": "0600", 
                "mtime": 1525151620.0765016, 
                "nlink": 1, 
                "path": "/live/persistence/TailsData_unlocked/persistence.conf", 
                "pw_name": "tails-persistence-setup", 
                "readable": false, 
                "rgrp": false, 
                "roth": false, 
                "rusr": true, 
                "size": 527, 
                "uid": 115, 
                "version": null, 
                "wgrp": false, 
                "woth": false, 
                "writeable": false, 
                "wusr": true, 
                "xgrp": false, 
                "xoth": false, 
                "xusr": false
            }
        }, 
        "msg": "All assertions passed"
    }
    ok: [localhost] => (item={'_ansible_parsed': True, u'stat': {u'isuid': False, u'uid': 115, u'exists': True, u'attr_flags': u'', u'woth': False, u'isreg': True, u'device_type': 0, u'mtime': 1525151620.0765016, u'block_size': 4096, u'inode': 12, u'isgid': False, u'size': 527, u'executable': False, u'roth': False, u'charset': u'unknown', u'readable': False, u'version': None, u'pw_name': u'tails-persistence-setup', u'gid': 122, u'ischr': False, u'wusr': True, u'writeable': False, u'isdir': False, u'blocks': 8, u'xoth': False, u'rusr': True, u'nlink': 1, u'issock': False, u'rgrp': False, u'gr_name': u'tails-persistence-setup', u'path': u'/live/persistence/TailsData_unlocked/persistence.conf', u'xusr': False, u'atime': 1525151594.3645008, u'mimetype': u'unknown', u'ctime': 1525151620.0765016, u'isblk': False, u'xgrp': False, u'dev': 65024, u'wgrp': False, u'isfifo': False, u'mode': u'0600', u'islnk': False, u'attributes': []}, u'changed': False, '_ansible_no_log': False, 'item': u'/live/persistence/TailsData_unlocked/openssh-client', '_ansible_item_result': True, 'failed': False, u'invocation': {u'module_args': {u'checksum_algorithm': u'sha1', u'get_checksum': True, u'follow': False, u'path': u'/live/persistence/TailsData_unlocked/persistence.conf', u'get_md5': True, u'get_mime': True, u'get_attributes': True}}, '_ansible_ignore_errors': None}) => {
        "changed": false, 
        "item": {
            "changed": false, 
            "failed": false, 
            "invocation": {
                "module_args": {
                    "checksum_algorithm": "sha1", 
                    "follow": false, 
                    "get_attributes": true, 
                    "get_checksum": true, 
                    "get_md5": true, 
                    "get_mime": true, 
                    "path": "/live/persistence/TailsData_unlocked/persistence.conf"
                }
            }, 
            "item": "/live/persistence/TailsData_unlocked/openssh-client", 
            "stat": {
                "atime": 1525151594.3645008, 
                "attr_flags": "", 
                "attributes": [], 
                "block_size": 4096, 
                "blocks": 8, 
                "charset": "unknown", 
                "ctime": 1525151620.0765016, 
                "dev": 65024, 
                "device_type": 0, 
                "executable": false, 
                "exists": true, 
                "gid": 122, 
                "gr_name": "tails-persistence-setup", 
                "inode": 12, 
                "isblk": false, 
                "ischr": false, 
                "isdir": false, 
                "isfifo": false, 
                "isgid": false, 
                "islnk": false, 
                "isreg": true, 
                "issock": false, 
                "isuid": false, 
                "mimetype": "unknown", 
                "mode": "0600", 
                "mtime": 1525151620.0765016, 
                "nlink": 1, 
                "path": "/live/persistence/TailsData_unlocked/persistence.conf", 
                "pw_name": "tails-persistence-setup", 
                "readable": false, 
                "rgrp": false, 
                "roth": false, 
                "rusr": true, 
                "size": 527, 
                "uid": 115, 
                "version": null, 
                "wgrp": false, 
                "woth": false, 
                "writeable": false, 
                "wusr": true, 
                "xgrp": false, 
                "xoth": false, 
                "xusr": false
            }
        }, 
        "msg": "All assertions passed"
    }
    ok: [localhost] => (item={'_ansible_parsed': True, u'stat': {u'isuid': False, u'uid': 115, u'exists': True, u'attr_flags': u'', u'woth': False, u'isreg': True, u'device_type': 0, u'mtime': 1525151620.0765016, u'block_size': 4096, u'inode': 12, u'isgid': False, u'size': 527, u'executable': False, u'roth': False, u'charset': u'unknown', u'readable': False, u'version': None, u'pw_name': u'tails-persistence-setup', u'gid': 122, u'ischr': False, u'wusr': True, u'writeable': False, u'isdir': False, u'blocks': 8, u'xoth': False, u'rusr': True, u'nlink': 1, u'issock': False, u'rgrp': False, u'gr_name': u'tails-persistence-setup', u'path': u'/live/persistence/TailsData_unlocked/persistence.conf', u'xusr': False, u'atime': 1525151594.3645008, u'mimetype': u'unknown', u'ctime': 1525151620.0765016, u'isblk': False, u'xgrp': False, u'dev': 65024, u'wgrp': False, u'isfifo': False, u'mode': u'0600', u'islnk': False, u'attributes': []}, u'changed': False, '_ansible_no_log': False, 'item': u'/home/amnesia/Persistent/securedrop', '_ansible_item_result': True, 'failed': False, u'invocation': {u'module_args': {u'checksum_algorithm': u'sha1', u'get_checksum': True, u'follow': False, u'path': u'/live/persistence/TailsData_unlocked/persistence.conf', u'get_md5': True, u'get_mime': True, u'get_attributes': True}}, '_ansible_ignore_errors': None}) => {
        "changed": false, 
        "item": {
            "changed": false, 
            "failed": false, 
            "invocation": {
                "module_args": {
                    "checksum_algorithm": "sha1", 
                    "follow": false, 
                    "get_attributes": true, 
                    "get_checksum": true, 
                    "get_md5": true, 
                    "get_mime": true, 
                    "path": "/live/persistence/TailsData_unlocked/persistence.conf"
                }
            }, 
            "item": "/home/amnesia/Persistent/securedrop", 
            "stat": {
                "atime": 1525151594.3645008, 
                "attr_flags": "", 
                "attributes": [], 
                "block_size": 4096, 
                "blocks": 8, 
                "charset": "unknown", 
                "ctime": 1525151620.0765016, 
                "dev": 65024, 
                "device_type": 0, 
                "executable": false, 
                "exists": true, 
                "gid": 122, 
                "gr_name": "tails-persistence-setup", 
                "inode": 12, 
                "isblk": false, 
                "ischr": false, 
                "isdir": false, 
                "isfifo": false, 
                "isgid": false, 
                "islnk": false, 
                "isreg": true, 
                "issock": false, 
                "isuid": false, 
                "mimetype": "unknown", 
                "mode": "0600", 
                "mtime": 1525151620.0765016, 
                "nlink": 1, 
                "path": "/live/persistence/TailsData_unlocked/persistence.conf", 
                "pw_name": "tails-persistence-setup", 
                "readable": false, 
                "rgrp": false, 
                "roth": false, 
                "rusr": true, 
                "size": 527, 
                "uid": 115, 
                "version": null, 
                "wgrp": false, 
                "woth": false, 
                "writeable": false, 
                "wusr": true, 
                "xgrp": false, 
                "xoth": false, 
                "xusr": false
            }
        }, 
        "msg": "All assertions passed"
    }
    
    PLAY [Add FPF apt repository and install base packages.] ***********************************************************************
    
    TASK [Gathering Facts] *********************************************************************************************************
    ok: [app]
    fatal: [mon]: UNREACHABLE! => {"changed": false, "msg": "SSH Error: data could not be sent to remote host \"10.20.3.2\". Make sure this host can be reached over ssh", "unreachable": true}
    
    NO MORE HOSTS LEFT *************************************************************************************************************
    
    NO MORE HOSTS LEFT *************************************************************************************************************
    	to retry, use: --limit @/home/amnesia/Persistent/securedrop/install_files/ansible-base/securedrop-prod.retry
    
    PLAY RECAP *********************************************************************************************************************
    app                        : ok=1    changed=0    unreachable=0    failed=0   
    localhost                  : ok=4    changed=0    unreachable=0    failed=0   
    mon                        : ok=0    changed=0    unreachable=1    failed=0   
    
    TASK: Gathering Facts -------------------------------------------------- 60.24s
    TASK: validate : Check for persistence volume. -------------------------- 2.51s
    TASK: Gathering Facts --------------------------------------------------- 1.68s
    TASK: validate : Confirm persistence volume is configured. -------------- 0.31s
    TASK: validate : Confirm host OS is Tails. ------------------------------ 0.07s
    
    Playbook finished: Fri May  4 23:18:21 2018, 5 total tasks. 0:01:04 elapsed. 
    
    TASK: Gathering Facts -------------------------------------------------- 60.25s
    TASK: validate : Check for persistence volume. -------------------------- 2.51s
    TASK: Gathering Facts --------------------------------------------------- 1.68s
    TASK: validate : Confirm persistence volume is configured. -------------- 0.31s
    TASK: validate : Confirm host OS is Tails. ------------------------------ 0.07s
    
    Playbook finished: Fri May  4 23:18:21 2018, 5 total tasks. 0:01:04 elapsed. 

Also, this error appears on STDOUT:

    ERROR (run with -v for more): Command '['/home/amnesia/Persistent/securedrop/install_files/ansible-base/securedrop-prod.yml', '--ask-become-pass']' returned non-zero exit status 2

## Comments

Related to #1432, the Gigabyte BRIX can successfully run vanilla Ubuntu and afaik the ansible script reports no errors on installing to the app server (after several tries and fails probably related to Tor latency issues). The BIOS settings are also a huge pain to access and their in-package manual is wrong about which keys to press to press to get into it.

**Comment by b-meson** [comment]
I also had the same issue testing `0.7rc3` on the `app` server when installing it. It took me about 4 hours because the `ssh app` portion of the installer would time out. I am not sure why the app server did this and not the `mon` server. Over the weekend, I attempted to SSH into the app server and it would fail repeatedly although I could access the journalist and source interface. 

However I am not sure if deprecating this hardware suggestion is the correct approach. I think this warrants further debugging.

**Comment by huertanix** [comment]
A small detail I should add: Before running any ansible scripts, the app server had to have Ubuntu installed on it twice because it wouldn't boot after the first install. I had until now assumed that this was because I removed the install media a tad early in the reboot process before the machine was lights out but now I'm not so sure. I'm going to try comparing the composition of the mon and app hard drive (GUID partition vs. MBR, etc) to see if there's some important difference I'm missing.

**Comment by huertanix** [comment]
Update: I tried re-running just the partitioning step of the Ubuntu install and after cancelling the rest of the install to see if I could work around the repair step by just running the part where the bootloader is loaded to the MBR. It booted successfully once, then after rebooting again, it only powers on but does not display anything. No grub bootloader, no BIOS disk error. Nothing but a blank empty screen.

**Comment by huertanix** [comment]
Update: I ended up reformatting the drive, but in this case with a GUID partition scheme instead of legacy MBR. I still followed the documentation's instructions and installed to MBR though. Also: I reset the BIOS settings on the BRIX to their default factory settings (I was able to press the Delete key to get in only after removing the hard drive with a bootable Ubuntu drive plugged in). This worked, survived a reboot, and I was able to continue with the SD install successfully.
